pandas/tests/arrays/categorical/test_replace.py

```diff
@@ -78,3 +78,13 @@ def test_replace_categorical_ea_dtype():
     result = pd.Series(cat).replace(["a", "b"], ["c", pd.NA])._values
     expected = Categorical(pd.array(["c", pd.NA], dtype="string"))
     tm.assert_categorical_equal(result, expected)
+
+
+def test_replace_maintain_ordering():
+    # GH51016
+    dtype = pd.CategoricalDtype([0, 1, 2], ordered=True)
+    ser = pd.Series([0, 1, 2], dtype=dtype)
+    result = ser.replace(0, 2)
```

Comment:
Do we have a test where a missing value is replaced?

Reply:
I don't see any. `Series(category).replace` doesn't actually work for replacing missing values - see below. I checked 1.5 and it doesn't replace in 1.5 either.

```
In [13]: import pandas as pd

In [14]: import numpy as np

In [15]: dtype = pd.CategoricalDtype(["A", "B"])

In [16]: ser = pd.Series(["A", None], dtype=dtype)

In [17]: ser
Out[17]: 
0      A
1    NaN
dtype: category
Categories (2, object): ['A', 'B']

In [18]: ser.replace(np.nan, "C")           # <- this doesn't actually replace the missing value
Out[18]: 
0      A
1    NaN
dtype: category
Categories (2, object): ['A', 'B']
```